cmake/sphinx.cmake

```diff
@@ -0,0 +1,107 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Part of the Tit Solver project, under the MIT License.
+# See /LICENSE.md for license information. SPDX-License-Identifier: MIT
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+include_guard()
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Find the Sphinx executable.
+find_program(SPHINX_BUILD_EXE NAMES "sphinx-build" REQUIRED)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+##
+## Add a Sphinx documentation target.
+##
+function(add_sphinx_target)
+  # Parse and check arguments.
+  cmake_parse_arguments(SPHINX "" "NAME" "" ${ARGN})
+  if(NOT SPHINX_NAME)
+    message(FATAL_ERROR "Sphinx target name must be specified.")
+  endif()
+
+  # Check that the configuration file exists.
+  if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/conf.py")
+    message(FATAL_ERROR "Sphinx configuration file does not exist!")
+  endif()
+
+  # Find all source files: `*.rst` files for documentation,
+  # `*.html`, `*.css`, `*.js` and `*.svg` files for the theme.
+  set(SPHINX_SOURCES "conf.py")
+  foreach(EXT ".rst" ".html" ".css" ".js" ".svg")
+    file(GLOB_RECURSE FILES "${CMAKE_CURRENT_SOURCE_DIR}/*${EXT}")
+    list(APPEND SPHINX_SOURCES ${FILES})
+  endforeach()
+
+  # Create a stamp.
+  set(STAMP "${CMAKE_CURRENT_BINARY_DIR}/${SPHINX_NAME}.sphinx_stamp")
```

Comment:
Instead of a stamp, we need to generate a list of the output files by scanning the source directory.